frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -1,24 +1,42 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues } from 'formik';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import { safeDump } from 'js-yaml';
 import { ItemSelectorField } from '@console/shared';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-import { NormalizedEventSources } from '../import-types';
-import { getEventSourceData } from '../../../utils/create-eventsources-utils';
+import { NormalizedEventSources, EditorType } from '../import-types';
+import {
+  getEventSourceData,
+  getEventSourceYaml,
+  isKnownEventSourceSelected,
+} from '../../../utils/create-eventsources-utils';
 
 interface EventSourcesSelectorProps {
   eventSourceList: NormalizedEventSources;
 }
 
 const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSourceList }) => {
   const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
+  const [namespaceField] = useField('project.name');
   const handleItemChange = React.useCallback(
     (item: string) => {
-      const name = `data.${item.toLowerCase()}`;
-      setFieldValue(name, getEventSourceData(item.toLowerCase()));
-      setFieldTouched(name, true);
+      if (isKnownEventSourceSelected(item)) {
```

Comment:
I think we shouldn't have it here , maybe it's good to introduce a prop on this component `onChange` and handle these logic in callback in `eventSourceForm`

Reply:
done